dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,40 +52,177 @@ public Pattern getParameterPattern() {
         return paramRegex;
     }
 
-    public Map<String, DbParameterAccessor> getAllColumns(
-            final String tableOrViewName) throws SQLException {
-        String qry = "SELECT COLUMNNAME, COLUMNDATATYPE "
-                + "FROM SYS.SYSCOLUMNS WHERE REFERENCEID = "
-                + "(SELECT TABLEID FROM SYS.SYSTABLES WHERE TABLENAME = ?)";
-        return readIntoParams(tableOrViewName.toUpperCase(), qry);
+    public Map<String, DbParameterAccessor> getAllColumns(String tableName)
+            throws SQLException {
+        return new DatabaseTable(new DatabaseObjectName(getConnection().getSchema(),
+                                 tableName.toUpperCase())).getParams();
```

Comment:
Here the the two parameters of `DatabaseObjectName` are not quite aligned. I would approach that maybe like that:
``` java
        return new DatabaseTable(new DatabaseObjectName(
            getConnection().getSchema(), tableName.toUpperCase())).getParams();
// or
        return new DatabaseTable(new DatabaseObjectName(
            getConnection().getSchema(),
            tableName.toUpperCase())).getParams();
```